### DISCUSSION TITLE
How can I add a remote worker when DS deploy on K8s

### DISCUSSION THREAD

**Discussion by batmanneverdie** [apache/dolphinscheduler#14249] [Q&A]
ds version: 3.1.3
master & worker deploy on k8s with `headless`.

If add a remote worker which don't deploy on k8s, then the remote worker will register Ip to zookeeper, but the k8s master and worker register the pod name like this:
![image](https://github.com/apache/dolphinscheduler/assets/41139080/d0d56015-891b-41dc-a4ce-168c3df0a6c8)

How cloud remote worker communicate with the master?

What's the correct deployment type on K8s when have remote worker?

**Comment by fuchanghai** [comment]
k8s 注册的时候 指定了hostname 注册，你可以稍微改一下源码就好了

**Reply by batmanneverdie** [reply]
已修改了相关代码。
